src/VisualStudio/CSharp/Impl/CSharpPackage.cs

```diff
@@ -22,6 +22,36 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.CSharp.LanguageService
 {
+    // The option page configuration is duplicated in PackageRegistration.pkgdef.
```

Comment:
How can we ensure these don't get out of sync? Because it'd be pretty easy for this comment to get missed...

Reply:
We don't right now.

Reply:
We could write a build task that generates the pkgdef entries based on the attributes. But I'd rather VS to not require both.

Reply:
I guess this PR makes me regret the pkgdef generator removal entirely, and this PR isn't really putting the codebase in a state that I'd want to keep it in. If we take this as a tactical fix, what's the long term plan forward?

Also, have we audited to confirm these are the only attributes where this is a problem? Or do we have further ones we have to put back as well?

Reply:
How do you audit that? I can't tell if some component is reflecting over our package types and looking for some attributes. We'll have to see if anything else breaks.

The long term plan is to work with VS platform on a fix. They can, e.g. avoid reflecting over these attributes and read all information from registry.

Reply:
Filed https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1121641

Reply:
@tmat Did we ask the SDK team if they knew of any others?

Reply:
I'll follow up on that. If we identify more attributes that are needed we can add them in a follow up PR. I don't think we need to block this one until we determine all of them.